### PR TITLE
Remove reversible masking

### LINKED ISSUE #28776: Harmonise Masking Approaches, removing reversible masking
The cunfusion caused by having two masking approaches was discussed at the TSC 02/06/2020, and it was agreed that we should harmonise on the more widely used non-reversible masking.  Please see the minutes for more details.

The tasks in this issue are:
1. Check with the ESS and ILL whether this will cause problems as they were not present at the meeting. (@gvardany , @SimonHeybrock please let me know if this will cause problems with your established workflows).
1. Adjust MaskDetectorsIf and MaskInstrument to clear the data after masking (update the docs too)
1. Remove the ClearMaskedDetectors algorithm.

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**
1. MaskInstrument and MaskDetectorsIf changed to clear masked spectra after applying masking
1. Algorithm ClearMaskedSpectra Removed as its only purpose has been replaced in point 1 above
1. Updates to the workflow in ISIS SANS and AlignAndFocusPowder to replace MaskInstrument + ClearMaskedSpectra with just MaskInstrument.
1. Deprecated MaskInstrument as MaskDetectors should be used instead.

**To test:**

1. Unit tests and doc tests must pass (that covers most of it)
1. Code review

Fixes #28776

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/sans/algorithm_detail/mask_workspace.py [hunk 1]
```diff
@@ -138,7 +137,7 @@ def mask_with_mask_files(mask_info, workspace):
             load_alg.execute()
             masking_workspace = load_alg.getProperty("OutputWorkspace").value
             # Could use MaskDetectors directly with masking_workspace but it does not
-            # support MPI. Use a three step approach via a, b, and c instead.
+            # support MPI. Use a three step approach via a and  b instead.
```
Comment (ConorMFinn) [inline scripts/SANS/sans/algorithm_detail/mask_workspace.py:140]:
```suggestion
            # support MPI. Use a two step approach via a and b instead.
```

File: docs/source/release/v5.1.0/framework.rst [hunk 2]
```diff
@@ -39,9 +39,15 @@ Algorithms
    When such incomplete data is encountered, it is skipped until the next valid data is encountered and a
    warning is printed at algorithm completion of the total number of data bytes discarded.
 - A bug introduced in v5.0 causing error values to tend to zero on multiple instances of :ref:`Rebin2D <algm-Rebin2D>` on the same workspace has been fixed.
+- A form of reversible masking that could lead to misleading and incorrect results has been removed from Mantid,
+  this means that ClearMaskedSpectra is no longer necessary after calling :ref:`MaskInstrument <algm-MaskInstrument>`
+  and :ref:`MaskDetectorsIf <algm-MaskDetectorsIf>`.
+  ClearMaskedSpectra has been Removed as it no longer has a use,
```
Comment (ConorMFinn) [inline docs/source/release/v5.1.0/framework.rst:45]:
```suggestion
  ClearMaskedSpectra has been removed as it no longer has a use,
```

**Review by cailafinn — CHANGES_REQUESTED** [review]
Changes make the masking algorithms much more intuitive. Just a couple of incredibly minor grammar nitpicks

**Comment by SimonHeybrock** [comment]
Question: What about saved workspaces (prior this change) loaded back now. Do masked spectra get cleared? Should they?

**Comment by NickDraper** [comment]
No we will not alter saved workspaces, I think this would be an unexpected change.  Just the algorithms will now clear the data as well as apply the mask flag.

**Comment by SimonHeybrock** [comment]
> No we will not alter saved workspaces, I think this would be an unexpected change. Just the algorithms will now clear the data as well as apply the mask flag.

But then `ClearMaskedSpectra` should not be removed, otherwise users will not be able to clear that data?

**Comment by NickDraper** [comment]
The only uses I have found in the code base have been immediate clearing of the data after using MaskInstrument.  So these will not cause a problem.  Are you aware of anyone actually saving uncleared data in files, or is this a theoretical problem?

**Comment by SimonHeybrock** [comment]
I am not aware of anyone doing this, no.

**Review by cailafinn — APPROVED** [review]
Looks good to me now. This makes a lot more sense than the old workflow. Nice one 👍

**Pull request merged**